### ISSUE TITLE
Feature: Force rearming plane to leave airpad

### ISSUE THREAD

**Issue by thomas11111111** [ZeroK-RTS/Zero-K#3971]
The feature allows a plane to move off the airpad it is rearming itself at, without waitting for the rearm to complete.
Issued as a modified move command alt+right click.
The plane takes of, relocates to the end of the move command and does not queue a rearm after the move command.
Issued with space+move, the plane takes takes off moves away from the pad to the end of the move command and returns to the same pad.
Issued with ctrl+alt+move, the plane moves away from the pad to the end of the move command and queues a rearm command to the nearest available pad.

**Comment by GoogleFrog** [comment]
All of these modkey combinations are already in use. I don't think I would want to use them for this even if they were not in use. I prefer units to be stuck to the rearm pad rather than them requiring an obscure UI to remove. If there is no way to remove them then no player is missing out by not knowing the UI.

How about something like selection rank. If you give an order and your selection consists entirely of planes on rearm pads, then the planes are told to leave the pad. Technically this could be implemented as a 'leave pad' parameter-less command that is automatically issued by the UI in the appropriate situation (as well as being bindable in hotkeys).

I believe my suggestion to be superior because it should automatically do what people want, i.e. when they tell a single rearming bomber to move off the pad it will do so without the player requiring any special knowledge. Requiring a modkey for a basic function is terrible.

**Comment by Gyslain** [comment]
I lost 4 lichos stuck on airpad. I resigned after that.

**Comment by Sharkgamedev** [comment]
#4136 With what I'm currently working on it wouldn't be too much for me to have an exclude airpad command issued if you move the plane off of an airpad so they will not go back to there, the question is should it be temporary and for all bombers or just the ones you moved?

**Comment by mojjj** [comment]
lost so many lichos to skythes killing them while rearming. should not be require to self-d the pad.
give the pad a d-gun button to send all planes away?

**Comment by GoogleFrog** [comment]
A D sounds ok too.

**Comment by Sharkgamedev** [comment]
But with a dgun is there a set time that it takes effect? Would it simply get the planes currently on there off but leave others to be able to land immediately? What if this was a modification/expansion of airpad exclusion where planes in an airpad when excluded would skedaddle instead of waiting. Or would a simple dgun burst like you talked about be better for the uses you imagine?

**Comment by sprunk** [comment]
Maybe bombers could avoid landing on pads on Wait (and possibly D would also toggle Wait)?

**Comment by Sharkgamedev** [comment]
> Maybe bombers could avoid landing on pads on Wait (and possibly D would also toggle Wait)?

That's what exclude airpads does which is why I'm mentioning it. I think it should probably go one way or the other with d pad command/wait inheriting exclude airpads functionality and get rid of the current command or just keep the current airpad exclude command.

**Comment by sprunk** [comment]
Right, I forgot that exists. In that case how about doing what GF suggested above (when all selected planes are reloading, they can leave via any regular order, otherwise they ignore it) and marking pads vacated this way as excluded?

**Comment by Sharkgamedev** [comment]
That sounds good! Perhaps we should wait and see if googlefrog has anything to add.

**Comment by GoogleFrog** [comment]
I think I have said everything. I'll remind everyone that unit selection information has no place in the sim. So

> Technically this could be implemented as a 'leave pad' parameter-less command that is automatically issued by the UI in the appropriate situation (as well as being bindable in hotkeys).

Now that we have command hiding it could even be a full command, but not visible on the base UI by default.

**Comment by mojjj** [comment]
bump

https://zero-k.info/Battles/Detail/1604660

**Comment by Sharkgamedev** [comment]
Ah, yeah I'll take another look at this and that old PR in another month or so if nobody else wants to

**Comment by mojjj** [comment]
could this solution be applied to all plane rearming platforms? reef, air factory, airpad.

thank you for your effort!

**Comment by Sharkgamedev** [comment]
> Right, I forgot that exists. In that case how about doing what GF suggested above (when all selected planes are reloading, they can leave via any regular order, otherwise they ignore it) and marking pads vacated this way as excluded?

One question here, is it really wise to 'mark pads vacated this way as excluded' since the exclusion command is not enabled by default, how would players know how to un-exclude it in the future? Perhaps that's an argument to transitioning exclusion functionality to the wait command?

**Comment by amnykon** [comment]
With https://github.com/ZeroK-RTS/Zero-K/pull/5388 (add rearming override selection rank), I think we should consider allowing normal moves to let planes leave before they are armed.

**Comment by mojjj** [comment]
obligatory "any progress?" bump
